src/main/kotlin/com/demonwav/mcdev/util/McBytecodeUtil.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+@file:JvmName("McBytecodeUtil")
+package com.demonwav.mcdev.util
+
+import com.intellij.psi.PsiArrayType
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiClassType
+import com.intellij.psi.PsiField
+import com.intellij.psi.PsiMethod
+import com.intellij.psi.PsiPrimitiveType
+import com.intellij.psi.PsiType
+import com.intellij.psi.util.TypeConversionUtil
+
+private const val INTERNAL_CONSTRUCTOR_NAME = "<init>"
+
+// Type
+
+internal val PsiPrimitiveType.internalName: Char
+    get() = when (this) {
+        PsiType.BYTE -> 'B'
+        PsiType.CHAR -> 'C'
+        PsiType.DOUBLE -> 'D'
+        PsiType.FLOAT -> 'F'
+        PsiType.INT -> 'I'
+        PsiType.LONG -> 'J'
+        PsiType.SHORT -> 'S'
+        PsiType.BOOLEAN -> 'Z'
+        PsiType.VOID -> 'V'
+        else -> throw IllegalArgumentException("Unsupported primitive type: $this")
+    }
+
+private fun PsiClassType.erasure() = TypeConversionUtil.erasure(this) as PsiClassType
+
+internal val PsiClassType.internalName
+    get() = erasure().resolve()!!.internalName
```

Comment:
I see your point. The reason I've made these properties throw an exception is mostly to simplify the code and to keep it consistent. There are a lot of other methods where this value is required to do certain checks or to build a part of the result (e.g. the descriptor generation for a specific method/field). For some of them I can't simply return null. 

For example, directly below this property we have the `appendInternalName` method which takes a `StringBuilder`, appends the internal name and returns the builder again. I cannot return null for this method, so we would have one method which returns null and one method which throws an exception.

It gets more complicated once we get to the descriptors. It also has a property and an append method, but at the end the implementation of both methods uses the `appendInternalName` method - how would we return null for the descriptor if one of the classes cannot be resolved?

**TL;DR:** The reason why I didn't make it return null is that, at the end, all code that is using these properties/methods requires the descriptors to continue successfully. In my testing I've never encountered a case where `resolve()` returned null. These cases would need special handling instead of just failing silently so throwing an exception was the better solution to allow us to handle these cases more leniently in the future.